spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin2/sender/RestTemplateSender.java

```diff
@@ -77,6 +81,16 @@ else if (this.encoding == Encoding.JSON) {
 		this.messageEncoder = BytesMessageEncoder.forEncoding(this.encoding);
 	}
 
+	private String buildUrlWithCustomPathIfNecessary(final String baseUrl,
+			final String customApiPath, final String defaultUrl) {
+		if (Objects.nonNull(customApiPath)) {
```

Comment:
Actually we could call `StringUtils.hasText(...)`

Reply:
@marcingrzejszczak , using nonNull was the intention. This is to cater cases where:

1. Null (when the apiPath is not configured at all, use the default
2. NonBlank (any text) as the custom api path
3. Blank (empty string) this is for cases no path needed at all (one example newrelic)

Reply:
1) `apiPath` should default to `api/v2/spans`  - that way we're simplifying the cases. 
2) Makes sense
3) Shouldn't it be then `/` ?

Reply:
@marcingrzejszczak 
We can't default to api/v2/spans in the apiPath because within RestTemplate the path is different between JSON2, PROTO and JSON. JSON is using v1. So I wanted to ensure that part of logic within the RestTemplateSender constructor being maintained.

For 3, at least for new relic case, there's no /. If you include /, tracing api will return 404. So for the path it must be empty.

Reply:
OK, now I understand! You're right :+1: